### PR TITLE
async actions type refactoring

### LINKED ISSUE #153: Improve AsyncActionCreator type signature to handle optional cancel property correctly
[Issuehunt badges](https://issuehunt.io/r/piotrwitek/typesafe-actions/issues/153)

## Is your feature request related to a real problem or use-case?
Fix cancel is never #152

## Describe a solution including usage in code example
Improve the resulting type signature of `AsyncActionCreator` making it more consistent with the rest of the API surface and to handle optional cancel property correctly.

The resulting signature must be looking like this:
```ts
AsyncActionCreator<["FETCH_USERS_REQUEST", undefined], ["FETCH_USERS_SUCCESS", User[]], ["FETCH_USERS_FAILURE", Error], ["FETCH_USERS_CANCEL", string]>
```

## Who does this impact? Who is this for?

All

## Describe alternatives you've considered (optional)

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Sponsors (Total: $40.00)

- [issuehunt issuehunt](https://issuehunt.io/u/issuehunt) ($40.00)

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/piotrwitek/typesafe-actions/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
</details>

### PR DESCRIPTION
**Pull request by maximelkin**
## Related issues:
- Resolved #153 

## Checklist

* [x] I have read [CONTRIBUTING.md](../CONTRIBUTING.md)
* [x] I have linked all related issues above
* [x] I have rebased my branch

For bugfixes:
* [ ] I have added at least one unit test to confirm the bug have been fixed
* [ ] I have checked and updated TOC and API Docs when necessary

For new features:
* [ ] I have added entry in TOC and API Docs
* [ ] I have added a short example in API Docs to demonstrate new usage
* [ ] I have added type unit tests with `dts-jest`
* [ ] I have added runtime unit tests with `dts-jest`

### CODE REVIEW

**Comment by piotrwitek** [comment]
Hey, thanks for your help although there are some issues with your PR.

https://github.com/piotrwitek/typesafe-actions/blob/master/CONTRIBUTING.md
point #2 is violated

**Comment by maximelkin** [comment]
Ok, sorry for violations and thanks for feedback

**Comment by piotrwitek** [comment]
@maximelkin that's alright, thanks for trying to help, although for now, I would strongly suggest to not work on this feature as there is a significant amount of changes in the next branch (currently heavily WIP) that is not compatible with your PR, I suggest to wait with any work until it's unblocked.

File: README.md [hunk 1]
```diff
@@ -702,22 +702,23 @@ _Create an object containing three enhanced action-creators to simplify handling
...
 
 ##### `AsyncActionCreator`
 
 ```ts
 type AsyncActionCreator<
-  [TRequestType, TRequestPayload],
-  [TSuccessType, TSuccessPayload],
-  [TFailureType, TFailurePayload],
-  [TCancelType, TCancelPayload]?
+  [TRequestType, TRequestPayload, TRequestMeta?],
```
Comment (piotrwitek) [inline README.md]: This 3rd parameter is definitely a no-go because it's not compatible with new v5 changes

File: README.md [hunk 2]
```diff
@@ -702,22 +702,23 @@ _Create an object containing three enhanced action-creators to simplify handling
...
+  [TSuccessType, TSuccessPayload, TSuccessMeta?],
+  [TFailureType, TFailurePayload, TFailureMeta?],
+  [TCancelType, TCancelPayload, TCancelMeta?]?
 > = {
-  request: StandardActionCreator<TRequestType, TRequestPayload>,
-  success: StandardActionCreator<TSuccessType, TSuccessPayload>,
-  failure: StandardActionCreator<TFailureType, TFailurePayload>,
-  cancel?: StandardActionCreator<TCancelType, TCancelPayload>,
+  request: StandardActionCreator<TRequestType, TRequestPayload, TRequestMeta>,
```
Comment (piotrwitek) [inline README.md]: Ditto StandardActionCreator

**Pull request closed without merging**